src/middlewared/middlewared/plugins/gluster_linux/volume.py

```diff
@@ -0,0 +1,569 @@
+from gfs.cli import volume
+from gfs.cli.utils import GlusterCmdException
+
+from middlewared.service import (CRUDService, accepts,
+                                 job, private, ValidationErrors,
+                                 CallError)
+from middlewared.schema import Dict, Str, Int, Bool, List
+
+from .utils import validate_gluster_jobs
+
+
+class GlusterVolumeService(CRUDService):
+
+    class Config:
+        namespace = 'gluster.volume'
+
+    def __volume_wrapper(self, method, args=None, kwargs=None):
+
+        result = b''
+
+        try:
+            if args and not kwargs:
+                result = method(*args)
+            if kwargs and not args:
+                result = method(**kwargs)
+            if args and kwargs:
+                result = method(*args, **kwargs)
+            if not args and not kwargs:
+                result = method()
+        except GlusterCmdException as e:
+            rc, out, err = e.args[0]
+            err = err if err else out
+            raise CallError(f'{err.decode().strip()}')
+        except Exception:
+            raise
+
+        if isinstance(result, bytes):
+            return result.decode().strip()
+
+        return result
+
+    @private
+    def common_validation(self, job):
+
+        verrors = ValidationErrors()
+
+        validate_gluster_jobs(self, verrors, job)
+
+    @private
+    def create_volume(self, data):
+
+        volname = data.pop('volname')
+        volbricks = data.pop('volbricks')
+
+        bricks = []
+        for i in volbricks:
+            peer = i.get('peer_name')
```

Comment:
`get` implies that key can be absent and we'll be happy with `None`, but we won't, `brick = peer + ':' + path` will crash if `peer` or `path` are `None`, and the problem is that exception you'll get will be less helpful than `KeyError`.

Reply:
@themylogin `peer_name` and `peer_path` will be enforced at schema level and require that those are populated with data.

Reply:
SO it should be `i['peer_name']` then

Reply:
thanks!